### PR TITLE
v0.28 dumps

### PR DESCRIPTION
**Pull request by dichotommy**
Changes related to the dump API in v0.28. For a full description of these changes, see the [associated issue](https://github.com/meilisearch/documentation/issues/1703).

To do:

- [ ] Fix links (only if necessary for merge, as the links are not actually broken, just dependent on changes made in another PR)

### CODE REVIEW

File: reference/api/dump.md [hunk 1]
```diff
@@ -28,35 +28,12 @@ The task queue will not process any further tasks during dump creation, but you
 
 ```json
 {
-  "uid": "20200929-114144097",
-  "status": "in_progress",
-  "startedAt": "2020-09-29T11:41:44.392327Z"
+  "taskUid": 0,
```
Comment (maryamsulemani97) [inline reference/api/dump.md:31]:
```suggestion
  "taskUid": 1,
```

I think we agreed to set any `taskUid` that wasn't creating an index to 1

File: reference/api/dump.md [hunk 2]
```diff
@@ -28,35 +28,12 @@ The task queue will not process any further tasks during dump creation, but you
...
+  "indexUid": null,
```
Comment (maryamsulemani97) [inline reference/api/dump.md:32]: We should either add a note here or add this to the 'Understanding tasks"  and "Summarized `task` Object for `202 Accepted`" tables in `learn/advanced/asynchronous_operations.md` that the `indexUid` will be `null` when task type is `dumpCreation`

File: learn/advanced/asynchronous_operations.md [hunk 3]
```diff
@@ -142,11 +142,7 @@ Had the task failed, the response would have included an `error` object:
 3. Once the task has completed processing, Meilisearch marks it as `succeeded`, if it was successful, or `failed`, if there was an error.
 4. Tasks marked as `succeeded` or `failed` are not deleted and will remain visible in [the task list](/reference/api/tasks.md#get-all-tasks)
 
-### Dumps
-
-While dumps and tasks are both asynchronous operations, they use separate queues and behave differently. For instance, creating a new dump will freeze the task queue until the dump has been generated.
-
-[You can read more about dumps in our dedicated guide.](/learn/advanced/dumps.md)
+Tasks are processed in the order they were enqueued, with one exception: `dumpCreation`. Dumps are prioritized over all other tasks in the queue. Their `uid` still represents when they were enqueued relative to other tasks.
```
Comment (maryamsulemani97) [inline learn/advanced/asynchronous_operations.md:145]: >Their `uid` still represents when they were enqueued relative to other tasks.

Do you think we should clarify that `uid` means task `uid` here?

File: reference/api/dump.md (hunk 1 above)
Comment (dichotommy) [reply reference/api/dump.md:31]: FWIW I think that specific discussion was about the quick start, where we actually have some knowledge of how many tasks the user has created up to that point; here, we don't. Still, this rule doesn't seem harmful so I personally am fine with it.

File: reference/api/dump.md [hunk 4]
```diff
@@ -6,19 +6,19 @@ Creating a dump is also referred to as exporting it, whereas launching Meilisear
...
 
 <RouteHighlighter method="POST" route="/dumps"/>
 
-Triggers a dump creation process. Once the process is complete, a dump is created in the [dumps directory](/learn/configuration/instance_options.md#dumps-destination). If the dumps directory does not exist yet, it will be created.
-
-**Meilisearch only processes one dump at a time.** If you attempt to create a dump while another dump is still processing, Meilisearch will throw a [`dump_already_processing` error](/reference/api/error_codes.md#dump-already-processing).
+Triggers a dump creation task. Once the process is complete, a dump is created in the [dumps directory](/learn/configuration/instance_options.md#dumps-destination). If the dumps directory does not exist yet, it will be created.
 
-The task queue will not process any further tasks during dump creation, but you can still add new requests to the queue.
+Dump tasks take priority over all other tasks in the queue. This means that a newly created dump task will be processed as soon as the current task is finished. [Learn more about asynchronous operations](/learn/advanced/asynchronous_operations.md).
```
Comment (maryamsulemani97) [inline reference/api/dump.md:21]:
```suggestion
Dump tasks take priority over all other tasks in the queue. This means that a newly created dump task will be processed as soon as the current task is finished.

[Learn more about asynchronous operations](/learn/advanced/asynchronous_operations.md).
```

**Review by maryamsulemani97 — APPROVED** [review]
🧚🏻‍♀️

**Comment by maryamsulemani97** [comment]
bors merge

**Comment by maryamsulemani97** [comment]
bors merge

**Pull request merged**